src/box/sql/insert.c

```diff
@@ -444,61 +444,6 @@ sqlInsert(Parse * pParse,	/* Parser context */
 		 */
 		if (trigger != NULL || vdbe_has_space_read(pParse, space_def))
 			useTempTable = 1;
```

Comment:
I suggest to move this if to your new if, for example:
```Diff

--- a/src/box/sql/insert.c
+++ b/src/box/sql/insert.c
@@ -442,8 +442,6 @@ sqlInsert(Parse * pParse,   /* Parser context */
                 * the SELECT statement. Also use a temp table in
                 * the case of row triggers.
                 */
-               if (trigger != NULL || vdbe_has_space_read(pParse, space_def))
-                       useTempTable = 1;
        } else {
                /* This is the case if the data for the INSERT is coming from a
                 * single-row VALUES clause
@@ -482,7 +480,9 @@ sqlInsert(Parse * pParse,   /* Parser context */
                goto insert_cleanup;
        }
 
-       if (pSelect && useTempTable) {
+       if (pSelect && (trigger != NULL ||
+                       vdbe_has_space_read(pParse, space_def))) {
+               useTempTable = 1;
                /* Invoke the coroutine to extract information from the SELECT
                 * and add it to a transient table srcTab. The code generated
                 * here is from the 4th template:
```

This is not necessary, but I think it will be more structured this way. Note that you must also move the comment.